app/services/stripe_service/api/payments.rb

```diff
@@ -29,17 +30,30 @@ def create_preauth_payment(tx, gateway_fields)
           sharetribe_payer_id: tx.starter_id,
           sharetribe_mode: stripe_api.charges_mode(tx.community_id)
         }
-        stripe_charge = stripe_api.charge(
-          community: tx.community_id,
-          token: source_id,
-          seller_account_id: seller_id,
-          amount: total.cents,
-          fee: commission.cents + buyer_commission.cents,
-          currency: total.currency.iso_code,
-          description: description,
-          metadata: metadata)
-
-        payment = PaymentStore.create(tx.community_id, tx.id, {
+
+        if source_id.present?
+          stripe_charge = stripe_api.charge(
+            community: tx.community_id,
+            token: source_id,
+            seller_account_id: seller_id,
+            amount: total.cents,
+            fee: commission.cents + buyer_commission.cents,
+            currency: total.currency.iso_code,
+            description: description,
+            metadata: metadata)
+        elsif payment_method_id.present?
```

Comment:
Let's change the reporting here to log correctly whether this was a payment using payment intent or a charge. Currently we have `report.create_charge_start` and `report.create_charge_success`. We should have equivalent ones for payment intents. Likewise, let's support reporting when capturing is done via a payment intent.

When looking at this, I spotted that there wasn't reporting for cancelling in `cancel_preauth` function. Let's add that. We could report the charge ID / payment intent ID.

Reply:
@bladealslayer 
Changes applied. It leads to massive refactoring to avoid duplicate code.